From 78fd73064ca8e01b4c0984605e3d93291abdc659 Mon Sep 17 00:00:00 2001
Message-Id: <78fd73064ca8e01b4c0984605e3d93291abdc659.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak M <m.deepak@intel.com>
Date: Mon, 26 Jan 2015 01:24:13 +0530
Subject: [PATCH 18/95] MUST_REBASE [VPG]: video/adf/intel: Adding the PIPE C
 related planes

The PIPE C related planes namely tertiary plane, sprite E and F are
added in the plane options.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5544
Change-Id: I265fc7a49c9366c1e8fbad46f4e48d6a05616a4f
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_pri_plane.c |    3 +++
 drivers/video/adf/intel/core/vlv/vlv_sp_plane.c  |    8 ++++++++
 2 files changed, 11 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c b/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
index 5921062..35ba8c8 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pri_plane.c
@@ -162,6 +162,9 @@ static int init_context(struct vlv_pri_plane_context *ctx, u8 idx)
 	case SECONDARY_PLANE:
 		ctx->plane = 1;
 		break;
+	case TERTIARY_PLANE:
+		ctx->plane = 2;
+		break;
 	default:
 		return -EINVAL;
 	}
diff --git a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
index 1eed46c..b23f41e 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_sp_plane.c
@@ -128,6 +128,14 @@ static int context_init(struct vlv_sp_plane_context *ctx, u8 idx)
 		ctx->plane = 1;
 		ctx->pipe = 1;
 		break;
+	case SPRITE_E:
+		ctx->plane = 0;
+		ctx->pipe = 2;
+		break;
+	case SPRITE_F:
+		ctx->plane = 1;
+		ctx->pipe = 2;
+		break;
 	default:
 		return -EINVAL;
 	}
-- 
1.7.9.5

